internal/work/dispatcher.go

```diff
@@ -193,6 +193,12 @@ func (d *Dispatcher) Connect() error {
 }
 
 func (d *Dispatcher) Dispatch(data yggdrasil.Data) error {
+	var err error
+	data.Directive, err = ScrubName(data.Directive)
+	if err != nil {
+		log.Warn(err)
```

Comment:
That's true. I set it at a higher level in the hopes that a user would see it and it would encourage service developers to fix their directive names. That's probably wishful thinking though.

Reply:
Will be service developers able to change directive name? They will have to support old and new workers, right. Will they be able to distinguish, what yggdrasil/rhcd version is on the other side? The yggdrasil.service still send following message during connecting:

```json
{
    "type": "connection-status",
    "message_id": "41c09a98-cbe5-4b79-82a0-63493a617bfc",
    "response_to": "",
    "version": 1,
    "sent": "2024-03-22T09:24:33.864006624+01:00",
    "content": {
        "canonical_facts": null,
        "dispatchers": {},
        "state": "online"
    }
}
```

The version is still the same (1). Should not we bump up this version?

Reply:
Yes, service developers can change the directive name, but they need to be aware of old clients running the old directive name. They can distinguish between client versions because new clients (yggdrasil 0.3 and newer) [include the client version in the connection-status message](https://github.com/RedHatInsights/yggdrasil/commit/512591a68671ca957f85ed68b773efbfdaeecc45). The version field is currently unused, but was added for situations like this where we need to break protocol conformance. I don't think that this is the case here though; examining the [`content.client_version` field](https://github.com/RedHatInsights/yggdrasil/wiki/Protocol#connection-status) should be sufficient.